app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,66 @@
+class PaymentsController < ApplicationController
+
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @payments = current_company.payments
+  end
+
+  def new
+    @payment = Payment.new
+    find_invoice(params[:id])
+  end
+
+  def edit
+    find_invoice(@payment.invoice_id)
+    params[:id] = @payment.invoice_id
+  end
+
+  def create
+    @payment = Payment.new(payment_params)
+    find_invoice(params[:id])
+    if @payment.save
+      if @payment.invoice.total_amount_payments < @payment.invoice.total_amount
+        @payment.invoice.status = Invoice.statuses["partially paid"]
+      else
+        @payment.invoice.status = Invoice.statuses["paid"]
+      end
+      @payment.invoice.save
+      if(params[:email].present?)
```

Comment:
Avoid using parenthesis for simple conditions. Also if there is only one line of instruction to be executed then you can check condition inline.

eg. ` execute_me if condition`